sw/device/tests/dif/dif_csrng_smoketest.c

```diff
@@ -0,0 +1,37 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_csrng.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+const test_config_t kTestConfig;
+
+bool test_main() {
+  const dif_csrng_params_t params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR),
+  };
+  dif_csrng_t csrng;
+  CHECK(dif_csrng_init(params, &csrng) == kDifCsrngOk);
+
+  const dif_csrng_config_t config = {
+      .debug_config =
+          (const dif_csrng_debug_config_t){
```

Comment:
I don't think you need the cast here; this is one of the few places C will just infer the type for you IIRC?

Reply:
Done. Sorry, this is mostly just an old habit on my part. I will keep this in mind.

Reply:
Hehe. C is a confusing little language.